frontend/src/pages/landingPage/index.js

```diff
@@ -23,6 +25,22 @@ export default function LandingPage() {
 
     useEffect(() => {
         fetchItems();  // Fetch all items initially
+
+         // Check if user is logged in
+         const userToken = localStorage.getItem("token");
+         console.log('userToken', userToken)
+         if (userToken) {
+             // Fetch user profile details if a token is present
+             profileService.getProfile()
+             .then(response => {
+                console.log('response', response)
+                 setUser(response); // Set the user data to state
+             })
+             .catch(error => {
+                 console.error("Error fetching user details:", error);
+             });
+         }
```

Comment:
Great! this is what should be in the `fetchUser` function. we can either move this into that function and call the function here in the `useEffect`, all take the function out altogether.